tools/github_actions/attempt_merge_pr.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+# Script to attempt to merge an automatic PR from main to develop.
+
+# Exit script if you try to use an uninitialized variable.
+set -o nounset
+
+# Exit script if a statement returns a non-true return value.
+set -o errexit
+
+SOURCE_BRANCH=$1
+TARGET_BRANCH=$2
+GITHUB_TAGGING_TOKEN=$3
+GITHUB_USER="kedro-org"
+GITHUB_REPO="kedro"
+
+# Array of GitHub mergeable states that are valid to proceed with automatic PR merging.
+# Adding "unstable" can be convenient, as it allows to auto-merge pull requests that
+# pass just the required CI checks (whereas "clean" requires all CI checks to pass,
+# regardless of whether they are required or not). More info:
+# https://docs.github.com/en/graphql/reference/enums#mergestatestatus
+# https://github.com/octokit/octokit.net/issues/1763
+VALID_MERGEABLE_STATES=("clean" "unstable")
+
+find_github_pr() {
+  # Find a PR from source to target branch
+  # Returns PR number if GitHub returned exactly one such PR
+  endpoint="https://api.github.com/repos/${GITHUB_USER}/${GITHUB_REPO}/pulls?base=${TARGET_BRANCH}&head=${GITHUB_USER}:${SOURCE_BRANCH}&state=open"
+  response=$(curl --silent --header "Authorization: token ${GITHUB_TAGGING_TOKEN}" "${endpoint}")
+  num_open_prs=$(echo "$response" | tr '\r\n' ' ' | jq "length")
+  if [ "$num_open_prs" -eq 1 ]; then
+    echo "$response" | tr '\r\n' ' ' | jq ".[0].number"
+  fi
+}
+
+check_pr_mergeable() {
+  # Check that the given PR is in a mergeable state
+  pr=$1
+  endpoint="https://api.github.com/repos/${GITHUB_USER}/${GITHUB_REPO}/pulls/${pr}"
+  response=$(curl --silent --header "Authorization: token ${GITHUB_TAGGING_TOKEN}" "${endpoint}")
+  mergeable=$(echo "${response}" | tr '\r\n' ' ' | jq ".mergeable // false")  # default to false
+  echo "PR ${pr} mergeable: ${mergeable}"
+  mergeable_state=$(echo "${response}" | tr '\r\n' ' ' | jq --raw-output ".mergeable_state // \"unknown\"")
+  echo "PR ${pr} mergeable_state: ${mergeable_state}"
+  [ "${mergeable}" == true ] && [[ " ${VALID_MERGEABLE_STATES[@]} " =~ " ${mergeable_state} " ]]
```

Comment:
Shellcheck complains about this line:

```
$ shellcheck myscript
 
[Line 44:](javascript:setPosition(44, 36))
  [ "${mergeable}" == true ] && [[ " ${VALID_MERGEABLE_STATES[@]} " =~ " ${mergeable_state} " ]]
                                   ^-- [SC2199](https://www.shellcheck.net/wiki/SC2199) (error): Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).
>>                                                                     ^-- [SC2076](https://www.shellcheck.net/wiki/SC2076) (warning): Remove quotes from right-hand side of =~ to match as a regex rather than literally.
```